src/librustc/hir/map/mod.rs

```diff
@@ -347,28 +347,33 @@ impl<'hir> Map<'hir> {
                 };
                 DefKind::Ctor(ctor_of, def::CtorKind::from_hir(variant_data))
             }
-            Node::AnonConst(_)
-            | Node::Field(_)
-            | Node::Expr(_)
-            | Node::Stmt(_)
-            | Node::PathSegment(_)
-            | Node::Ty(_)
-            | Node::TraitRef(_)
-            | Node::Pat(_)
-            | Node::Binding(_)
-            | Node::Local(_)
-            | Node::Param(_)
-            | Node::Arm(_)
-            | Node::Lifetime(_)
-            | Node::Visibility(_)
-            | Node::Block(_)
-            | Node::Crate => return None,
+            Node::AnonConst(_) => DefKind::AnonConst,
+            Node::Field(_) => DefKind::Field,
+            Node::Expr(expr) => match expr.kind {
+                ExprKind::Closure { .. } => DefKind::ClosureExpr,
```

Comment:
Out of all expressions, only closures have `DefId`s. The `DefKind` could be named just `Closure`.